src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package duke;
+
+import duke.command.AliasCommand;
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.ExitCommand;
+import duke.command.PlaceCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyFieldDukeException;
+import duke.exception.InvalidCommandDukeException;
+import duke.exception.PlaceParseDukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+public class Parser {
+
+    /**
+     * Looks at first token, determine & construct duke.command.Command sub-classes as per parsed.
+     *
+     * @param str Expects string to be parse as duke.command.Command.
+     * @return duke.command.Command object as per parsed.
+     * @throws DukeException if command can't be parsed.
+     */
+    public static Command parse(String str) throws DukeException {
+        assert str != null;
+
+        str = str.trim();
+        int lastIndex = str.indexOf(' ');
+        if (lastIndex < 0) {
+            lastIndex = str.length();
+        }
+        String[] temp;
+        switch (str.substring(0, lastIndex)) {
+        case "bye":
+            return new ExitCommand();
+        case "delete":
+            return new DeleteCommand(str.substring(lastIndex));
+        case "done":
+            return new DoneCommand(str.substring(lastIndex));
+        case "find":
+            return new FindCommand(str.substring(lastIndex));
+        case "list":
+            return new ListCommand();
+        case "deadline":
+            temp = str.substring(8)
```

Comment:
Perhaps you can try not to use magic literal in your code as it will be more intuitive for the next developer.

Reply:
👍 Bad practice indeed.